### ISSUE TITLE
Batch subscribe unsubscribe

### ISSUE THREAD

**Issue by ga630sf** [brandonseydel/MailChimp.Net#360]
Hi Guys

Mailchimp api shows a Batch Subscribe/Unsubscribe members under their List API
https://developer.mailchimp.com/documentation/mailchimp/reference/lists/#create-post_lists_list_id

This looks much better than the actual Batch updates for members as it returns the results per member in the response rather than having to parse a json file
It is limited to 500 members per call however, but would still be better than Member Batch updates

Has this been implemented, or any plans to implement?

regards
Steve

**Comment by brandonseydel** [comment]
They added this more recently.  I will be adding this support soon.

**Comment by raduvremescu** [comment]
When you think it will be implemented (and some example).
10x.

**Comment by Telaran** [comment]
It's available since Version 4.0 (March 2019).

I did something like following code. Maybe there's a better approach (feel free if someone know one)

```CSharp
// Split the List to less than 500 Members
foreach (var chunkedMemberList in Chunkify(memberList, 450))
{
	var bList = new BatchList
	{
		UpdateExisting = true,
		Members = chunkedMemberList.ToList()
	};

	var partialResult = await mailChimpManager.Lists.BatchAsync(bList, mailchimpListId);
	foreach (var item in partialResult.Errors)
	{
		// Do Something with the failed imports
	}
}

// Function for split a list into smaller batches
public static IEnumerable<IEnumerable<T>> Chunkify<T>(this IEnumerable<T> source, int size)
{
	int count = 0;
	using (var iter = source.GetEnumerator())
	{
		while (iter.MoveNext())
		{
			var chunk = new T[size];
			count = 1;
			chunk[0] = iter.Current;
			for (int i = 1; i < size && iter.MoveNext(); i++)
			{
				chunk[i] = iter.Current;
				count++;
			}
			if (count < size)
			{
				Array.Resize(ref chunk, count);
			}
			yield return chunk;
		}
	}
}
```
